### PR TITLE
kubetest: initialize http.Transport only once and add unit test

### PR DESCRIPTION
**Pull request by ixdy**
Follow-up to #2394 from yesterday.

### CODE REVIEW

**Comment by ixdy** [comment]
/assign @fejta 

I thought I did that already.

File: kubetest/util_test.go [hunk 1]
```diff
@@ -255,3 +259,28 @@ func TestOutput(t *testing.T) {
 		}
 	}
 }
+
+func TestHttpFileScheme(t *testing.T) {
```
Comment (fejta) [inline kubetest/util_test.go]: Ooh unit tests! Very cool!

**Comment by fejta** [comment]
/lgtm

Extra kudos for including a unit test to cover the missing scenario

**Comment by ixdy** [comment]
(I also verified that if I remove the `RegisterProtocol`, the unit test fails.)

**Pull request merged**